### PR TITLE
fix the xpara error in new version of the pytest

### PR DESCRIPTION
**Pull request by Zheaoli**

### CODE REVIEW

File: pytest_xpara/plugin.py [hunk 1]
```diff
@@ -3,19 +3,35 @@
...
 
 
 def pytest_generate_tests(metafunc):
     if not metafunc.config.option.xpara:
         return
 
     try:
-        mark = metafunc.function.xparametrize
+        if hasattr(metafunc.function, "xparametrize"):
+            mark = metafunc.function.xparametrize
```
Comment (tonyseek) [inline pytest_xpara/plugin.py]: I prefer to use `getattr` and the `next(iter())` pattern instead. e.g:

``` python
mark = getattr(metafunc.function, 'xparametrize', None)
if mark is None:
    pmark = getattr(metafunc.function, 'pytestmark', None)
    if pmark is not None:
        mark = next(iter(x for x in pmark if x.name == 'xparametrize'), None)
```

File: pytest_xpara/plugin.py [hunk 2]
```diff
@@ -3,19 +3,35 @@
...
+        elif hasattr(metafunc.function, "pytestmark"):
+            temp_marks = [x for x in metafunc.function.pytestmark
+                          if x.name == "xparametrize"]
+            if len(temp_marks) > 0:
+                mark = temp_marks[0]
+            else:
+                return
+        else:
+            return
     except AttributeError:
```
Comment (tonyseek) [inline pytest_xpara/plugin.py:33]: Is this try-except block still necessary?

File: pytest_xpara/plugin.py [hunk 3]
```diff
@@ -69,7 +85,7 @@ def _load_data_as_yaml(current_dir, file_name):
     for ext_name in ('yaml', 'yml'):
         data_file = current_dir.join('%s.%s' % (file_name, ext_name))
         if data_file.exists():
-            return yaml.load(data_file.read())
+            return yaml.safe_load(data_file.read())
```
Comment (tonyseek) [inline pytest_xpara/plugin.py:86]: 👍🏼

File: setup.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 setup(
     name='pytest-xpara',
-    version='0.1.1',
+    version='0.1.2',
```
Comment (tonyseek) [inline setup.py]: Would it be a new minor release (`0.2.0`) since the Python 2.x will be out-of-support?

File: tox.ini [hunk 5]
```diff
@@ -14,3 +14,5 @@ commands =
     python setup.py check
     flake8
     py.test --cov={envsitepackagesdir}/pytest_xpara --cov-append {posargs}
+[run]
+parallel = True
```
Comment (tonyseek) [inline tox.ini]: Could you add a newline at end of file? I forgot to configure `.editorconfig` on this repository.

**Review by tonyseek — CHANGES_REQUESTED** [review]

File: setup.py (hunk 4 above)
Comment (Zheaoli) [reply setup.py]: Of course yes,

File: tox.ini (hunk 5 above)
Comment (Zheaoli) [reply tox.ini]: OK

File: pytest_xpara/plugin.py (hunk 2 above)
Comment (Zheaoli) [reply pytest_xpara/plugin.py:33]: I think one try-except block is enough

File: setup.py (hunk 4 above)
Comment (tonyseek) [inline setup.py]: 0.2.0

Comment (Zheaoli) [reply setup.py]: Done

**Review by tonyseek — APPROVED** [review]

**Pull request merged**